lib/sidekiq/job/iterable.rb

```diff
@@ -0,0 +1,260 @@
+# frozen_string_literal: true
+
+require_relative "iterable/enumerators"
+
+module Sidekiq
+  module Job
+    module Iterable
+      class Interrupted < ::RuntimeError; end
+
+      include Enumerators
+
+      # @api private
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      # @api private
+      module ClassMethods
+        def method_added(method_name)
+          if method_name == :perform
+            raise "Job that is using Iterable (#{self}) cannot redefine #perform"
+          end
+
+          super
+        end
+      end
+
+      # @api private
+      def initialize
```

Comment:
This module should include a `stopping?` method which checks `@done` on Sidekiq::Processor. Here we use a :lifecycle accessor to store the Processor on the Job.

```ruby
# In Sidekiq::Processor...
job = SomeJob.new
job.lifecycle = self if job.is_a?(Sidekiq::Job::Iteratable) # Processor
job.perform(*args)
job.stopping? # => job.lifecycle.stopping?
```

It's a little more complex but I'm trying to avoid any more global APIs like `Sidekiq.stopping?` to be more Ractor-friendly.